src/NServiceBus.AcceptanceTests/Sagas/When_a_finder_exists_and_found_saga.cs

```diff
@@ -41,13 +41,18 @@ public class CustomFinder : IFindSagas<TestSaga08.SagaData08>.Using<SomeOtherMes
                 // ReSharper disable once MemberCanBePrivate.Global
                 public Context Context { get; set; }
 
+                public ISagaPersister SagaPersister { get; set; }
+
                 public Task<TestSaga08.SagaData08> FindBy(SomeOtherMessage message, SynchronizedStorageSession storageSession, ReadOnlyContextBag context)
                 {
                     Context.FinderUsed = true;
-                    return Task.FromResult(new TestSaga08.SagaData08
+                    var sagaInstance = new TestSaga08.SagaData08
                     {
                         Property = "jfbsjdfbsdjh"
-                    });
+                    };
+                    //Make sure saga exists in the store because saving it in the end of the transaction extects it to be there.
```

Comment:
~~extects~~ expects 

Also, I'm not clear on what "saving it in the end of the transaction" means?